mssqlcli/packages/parseutils/tables.py

```diff
@@ -35,15 +35,21 @@ def extract_from_part(parsed, stop_at_punctuation=True):
                 for x in extract_from_part(item, stop_at_punctuation):
                     yield x
             elif stop_at_punctuation and item.ttype is Punctuation:
-                raise StopIteration
-            # An incomplete nested select won't be recognized correctly as a
-            # sub-select. eg: 'SELECT * FROM (SELECT id FROM user'. This causes
-            # the second FROM to trigger this elif condition resulting in a
-            # StopIteration. So we need to ignore the keyword if the keyword
-            # FROM.
-            # Also 'SELECT * FROM abc JOIN def' will trigger this elif
-            # condition. So we need to ignore the keyword JOIN and its variants
-            # INNER JOIN, FULL OUTER JOIN, etc.
+                # An incomplete nested select won't be recognized correctly as a
+                # sub-select. eg: 'SELECT * FROM (SELECT id FROM user'. This causes
+                # the second FROM to trigger this elif condition resulting in a
+                # StopIteration. So we need to ignore the keyword if the keyword
+                # FROM.
+                # Also 'SELECT * FROM abc JOIN def' will trigger this elif
+                # condition. So we need to ignore the keyword JOIN and its variants
+                # INNER JOIN, FULL OUTER JOIN, etc.
+                
+                # Unlikely in Python 2, StopIteration is converted into a RuntimeError
+                # when code runs 'raise StopIteration' in Python 3.
+                # Generators which explicitly raise StopIteration can generally be
+                # changed to simply return instead.
+                # (https://www.python.org/dev/peps/pep-0479/)
```

Comment:
Please add these details to the PR description, but I don't think we need these details in the code comments as it's pretty clear what the return does.

Reply:
removed.